File: Packs/CommonScripts/Scripts/SearchIndicator/SearchIndicator.py
import demistomock as demisto  # noqa: F401
from CommonServerPython import *  # noqa: F401


def search_indicators(args):
    # search for indicators
    indicators = demisto.executeCommand("findIndicators", {"query": args.get("query"), "size": args.get("size")})[0][
        "Contents"]

    # return specific information for found indicators
    filtered_indicators = []
    fields = ['id', 'indicator_type', 'value', 'score']
    if args.get("add_fields_to_context"):
        fields = fields + args.get("add_fields_to_context").split(",")
        fields = [x.strip() for x in fields]  # clear out whitespace
    for indicator in indicators:
        style_indicator = {}
        for field in fields:
            style_indicator[field] = indicator.get(field, indicator.get("CustomFields", {}).get(field, "n/a"))
        style_indicator["verdict"] = scoreToReputation(style_indicator['score'])
        filtered_indicators.append(style_indicator)

    headers = fields + ["verdict"]
    markdown = tableToMarkdown("Indicators Found", filtered_indicators, headers)
    return markdown, filtered_indicators


def main():
    args = demisto.args()
    try:
        readable_output, outputs = search_indicators(args)
        results = CommandResults(
            outputs_prefix='foundIndicators',
            outputs_key_field='id',
            readable_output=readable_output,
            outputs=outputs,
            ignore_auto_extract=True
        )
        return_results(results)
    except DemistoException as error:
        return_error(str(error), error)


if __name__ in ('__main__', '__builtin__', 'builtins'):
    main()

File: Packs/CommonScripts/Scripts/SearchIndicator/SearchIndicator.js
var res = executeCommand('findIndicators', {
    query: args.query,
    size: args.size
});

if (!isValidRes(res)) {
    if (res[0].Contents) {
        return {
            ContentsFormat: formats.markdown,
            Type: entryTypes.error,
            Contents: res[0].Contents
        };
    }
}

var filteredIndicators = []
if (
    res &&
    res[0] &&
    res[0].Contents
) {
    filteredIndicators = []

    fields = ["id", "indicator_type", "value", "score"]


    if (args.add_fields_to_context) {
        fields = fields.concat(args.add_fields_to_context.split(","));
        fields = fields.map(x => x.trim()); // clear out whitespace
    }

    for (var indicator of res[0].Contents) {
        var styleIndicator = {};
        for (var field of fields) {
            styleIndicator[field] =
                indicator[field] !== undefined ? indicator[field] :
                (indicator.CustomFields && indicator.CustomFields[field] !== undefined ?
                    indicator.CustomFields[field] :
                    "n/a");
        }

        styleIndicator["verdict"] = scoreToReputation(styleIndicator["score"])

        filteredIndicators.push(styleIndicator);
    }

    var headers = fields.concat(["verdict"]);

}

var ec = {
    'foundIndicators(val.id && val.id == obj.id)': filteredIndicators
};

return {
    Type: entryTypes.note,
    ReadableContentsFormat: formats.markdown,
    Contents: filteredIndicators,
    ContentsFormat: formats.json,
    HumanReadable: tableToMarkdown("Indicators Found", filteredIndicators, headers),
    EntryContext: ec,
    IgnoreAutoExtract: true
};